baseapp/baseapp.go

```diff
@@ -260,6 +260,18 @@ func DefaultStoreLoader(ms sdk.CommitMultiStore) error {
 	return ms.LoadLatestVersion()
 }
 
+// CommitMultiStore returns the root multi-store.
+// Extension snapshotter use this to read storage.
+func (app *BaseApp) CommitMultiStore() sdk.CommitMultiStore {
```

Comment:
maybe prefix comment with UNSAFE to users know not to use this for just anything. 

Im happy this is added, I was working on something that uses this

Reply:
It's for stuff like this, and not all extension snapshotter need the `cms`, I don't see a more elegant way to support this:

```
func NewApp() *App {
  ...
  app.SnapshotManager().RegisterExtension(
    NewWasmSnapshotter(app.CommitMultiStore(), wasmStoreKey),
  )
```

Reply:
I found a function like this useful for debugging app hash non-determinism sources.

I think it should be there but needs a more extensive comment.

Reply:
what do you think if we add:
```
	if app.sealed {
		panic("cannot call CommitMultiStore(): baseapp already sealed")
	}
```

Reply:
on second thought I have already started to use this for another feature and bez added it in on of his prs. Lets leave it as is.